test/libsolidity/SyntaxTestHelper.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/libsolidity/SyntaxTestHelper.h>
+#include <test/libsolidity/AnalysisFramework.h>
+#include <test/TestHelper.h>
+#include <boost/algorithm/string/replace.hpp>
+
+using namespace std;
+using namespace boost::unit_test;
+namespace fs = boost::filesystem;
+
+namespace dev
+{
+namespace solidity
+{
+namespace test
+{
+
+void SyntaxTester::runTest(SyntaxTest const& _test)
+{
+	vector<string> unexpectedErrors;
+	auto expectations = _test.expectations;
+	auto errorList = parseAnalyseAndReturnError(_test.source, true, true, true).second;
+
+	bool errorsMatch = true;
+
+	if (errorList.size() != expectations.size ())
+		errorsMatch = false;
+	else
+	{
+		for (size_t i = 0; i < errorList.size(); i++)
+		{
+			auto const& error = errorList[i];
+			auto const& expectation = expectations[i];
+			bool const typeMatches = error->typeName() == expectation.first;
+			bool const messageMatches = errorMessage(*error) == expectation.second;
+			if (!typeMatches || !messageMatches)
+			{
+				errorsMatch = false;
+				break;
+			}
+		}
+	}
+
+	if (!errorsMatch)
+	{
+		string msg = "Test expectation mismatch.\nExpected result:\n";
+		if (expectations.empty())
+			msg += "\tSuccess\n";
```

Comment:
To be pedantic, this should be "Success without warnings", but we can also keep it like that.

Reply:
Since the result of a "Success with warning(s)" will not be nor include "Success", but just be a list of warnings, I would say "Success" should be enough for the case without warnings.